src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -13054,6 +13006,48 @@ public virtual bool Equals(A other)
             var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseDll);
             comp.MakeTypeMissing(SpecialType.System_Boolean);
             comp.VerifyEmitDiagnostics(
+                // (2,1): error CS0518: Predefined type 'System.Boolean' is not defined or imported
+                // record A
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, @"record A
+{
+    public virtual bool Equals(A other)
+        => throw null;
+
+    System.Boolean System.IEquatable<A>.Equals(A x) => throw null;
+}").WithArguments("System.Boolean").WithLocation(2, 1),
```

Comment:
Can the scope of the error be reduced to the type name rather than the entire declaration?

Reply:
> Can the scope of the error be reduced to the type name rather than the entire declaration?

That would be rather difficult to do because I believe that error is coming from bound node factory that is using syntax node as location for the error to report. The name is not a node, but a token, I believe. Given that this is an edge case, if core library is missing, there would be more errors all over the place, I think the current behavior is acceptable. 

---
In reply to: [464575538](https://github.com/dotnet/roslyn/pull/46497#discussion_r464575538) [](ancestors = 464575538)